client/rest/src/plugins/metadata/metal.js

```diff
@@ -79,67 +77,57 @@ const metal = {
 			};
 		}
 
-		// Extract text section until null char is encountered.
-		const textBytes = [];
-		for (let i = 0; i < chunkData.chunkPayload.length && chunkData.chunkPayload[i]; i++)
-			textBytes.push(chunkData.chunkPayload[i]);
+		// Find the position of the null char.
+		const separatorIndex = chunkData.chunkPayload.indexOf(0);
 
 		return {
-			chunkPayload: new Uint8Array(chunkData.chunkPayload.subarray(textBytes.length + 1)),
-			chunkText: new Uint8Array(textBytes)
+			chunkPayload: chunkData.chunkPayload.subarray(separatorIndex + 1),
+			chunkText: chunkData.chunkPayload.subarray(0, separatorIndex)
 		};
 	},
 
 	/**
 	 * Decode binary data from chunks.
-	 * @param {Uint64} fitstKey ScopedMetadataKey of the beginning chunk.
+	 * @param {Uint64} firstKey ScopedMetadataKey of the beginning chunk.
 	 * @param {Array} chunks Chunk containing data to be decoded.
 	 * @returns {Buffer} Decoded binary data.
 	 */
-	decode(fitstKey, chunks) {
-		let scopedMetadataKey = fitstKey;
+	decode(firstKey, chunks) {
+		let scopedMetadataKey = firstKey;
 		let magic;
-		let decodedPayload = Buffer.from([]);
-		let decodedText = Buffer.from([]);
-		const findChunk = c => 0 === catapult.utils.uint64.compare(c.key, scopedMetadataKey);
+		const decodedPayload = [];
+		const decodedText = [];
```

Comment:
probably would use plurals in names since these are now arrays => decodedPayloads, decodedTexts

Reply:
🙏